feat: caching block hash

benesjan opened the pull request:
For [my other work](https://github.com/AztecProtocol/aztec-packages/pull/19030/changes) I need retrieval of a block header hash to be fast.

In this PR I ensure this is the case by caching the hash.

benesjan commented:
* **#19138** <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/19138?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.com/github/pr/AztecProtocol/aztec-packages/19138?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `next`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

dbanks12 reviewed (approved):
If that public tx simulator test suite passes after this, then this LGTM!

benesjan commented:
@dbanks12  that was fast. Thanks for the review!

benesjan committed:
feat: caching block hash

For [my other work](https://github.com/AztecProtocol/aztec-packages/pull/19030/changes) I need retrieval of a block header hash to be fast.

In this PR I ensure this is the case by caching the hash.
